packages/sns/test/consumers/SnsSqsPermissionConsumer.spec.ts

```diff
@@ -435,4 +441,71 @@ describe('SnsSqsPermissionConsumer', () => {
       })
     })
   })
+
+  describe('visibility timeout', () => {
+    const topicName = 'myTestTopic'
+    const queueName = 'myTestQueue'
+    let diContainer: AwilixContainer<Dependencies>
+
+    beforeEach(async () => {
+      diContainer = await registerDependencies({
+        permissionConsumer: asFunction(() => undefined),
+        permissionPublisher: asFunction(() => undefined),
+      })
+    })
+
+    afterEach(async () => {
+      await diContainer.cradle.awilixManager.executeDispose()
+      await diContainer.dispose()
+    })
+
+    it.each([false, true])('using 2 consumers with heartbeat -> %s', async (heartbeatEnabled) => {
+      let consumer1IsProcessing = false
+      let consumer1Counter = 0
+      let consumer2Counter = 0
+
+      const consumer1 = new SnsSqsPermissionConsumer(diContainer.cradle, {
+        creationConfig: {
+          topic: { Name: topicName },
+          queue: { QueueName: queueName, Attributes: { VisibilityTimeout: '2' } },
+        },
+        consumerOverrides: { heartbeatInterval: heartbeatEnabled ? 1 : undefined },
+        removeHandlerOverride: async () => {
+          consumer1IsProcessing = true
+          // wait for consumer2 to process message while this is still processing
+          await waitAndRetry(() => consumer2Counter > 0, 100, 30)
+          consumer1Counter++
+          consumer1IsProcessing = false
+          return { result: 'success' }
+        },
+      })
+      await consumer1.start()
+
+      const consumer2 = new SnsSqsPermissionConsumer(diContainer.cradle, {
+        locatorConfig: {
+          queueUrl: consumer1.subscriptionProps.queueUrl,
+          topicArn: consumer1.subscriptionProps.topicArn,
+          subscriptionArn: consumer1.subscriptionProps.subscriptionArn,
+        },
+        removeHandlerOverride: async () => {
+          consumer2Counter++
+          return { result: 'success' }
+        },
+      })
+      const publisher = new SnsPermissionPublisher(diContainer.cradle, {
+        locatorConfig: { topicArn: consumer1.subscriptionProps.topicArn },
+      })
+
+      await publisher.publish({ id: '10', messageType: 'remove' })
+      // wait for consumer1 to start processing to start second consumer
+      await waitAndRetry(() => consumer1IsProcessing, 5, 5)
+      await consumer2.start()
+
+      // wait for both consumers to process message
+      await waitAndRetry(() => consumer1Counter > 0 && consumer2Counter > 0, 100, 40)
```

Comment:
we should assert that it resolves to true and not just timeouts

Reply:
The check is done using counters, in the second `each` case I don't only want to know it is `false`, we should check that consumer1 was de only who received the message